File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.py
T = int(input())
for t in range(1, T + 1):
    N, M = map(int, input().split())
    arr = [list(map(int, input().split())) for _ in range(N)]
    max_flies = 0
    
    for i in range(N - M + 1):
        for j in range(N - M + 1):
            current_flies = 0
            for r in range(M):
                for c in range(M):
                    current_flies += arr[i + r][j + c]
            if current_flies > max_flies:
                max_flies = current_flies
                
    print(f"#{t} {max_flies}")

File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.java
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        for (int t = 1; t <= T; t++) {
            int N = sc.nextInt();
            int M = sc.nextInt();
            int[][] arr = new int[N][N];
            for (int i = 0; i < N; i++)
                for (int j = 0; j < N; j++)
                    arr[i][j] = sc.nextInt();
                    
            int maxFlies = 0;
            for (int i = 0; i <= N - M; i++) {
                for (int j = 0; j <= N - M; j++) {
                    int currentFlies = 0;
                    for (int r = 0; r < M; r++) {
                        for (int c = 0; c < M; c++) {
                            currentFlies += arr[i + r][j + c];
                        }
                    }
                    maxFlies = Math.max(maxFlies, currentFlies);
                }
            }
            System.out.println("#" + t + " " + maxFlies);
        }
        sc.close();
    }
}